runner/organizations_test.go

```diff
@@ -19,173 +19,544 @@ import (
+	s.Require().Nil(err)
+	s.Require().Equal(s.Fixtures.CreateOrgParams.Name, org.Name)
+	s.Require().Equal(s.Fixtures.Credentials[s.Fixtures.CreateOrgParams.CredentialsName].Name, org.CredentialsName)
+}
+
+func (s *OrgTestSuite) TestCreateOrganizationErrUnauthorized() {
+	_, err := s.Runner.CreateOrganization(context.Background(), s.Fixtures.CreateOrgParams)
+
+	s.Require().Equal(runnerErrors.ErrUnauthorized, err)
+}
+
+func (s *OrgTestSuite) TestCreateOrganizationEmptyParams() {
+	_, err := s.Runner.CreateOrganization(s.Fixtures.AdminContext, params.CreateOrgParams{})
+
+	s.Require().Regexp("validating params: missing org name", err.Error())
+}
+
+func (s *OrgTestSuite) TestCreateOrganizationMissingCredentials() {
+	s.Fixtures.CreateOrgParams.CredentialsName = "not-existent-creds-name"
+
+	_, err := s.Runner.CreateOrganization(s.Fixtures.AdminContext, s.Fixtures.CreateOrgParams)
+
+	s.Require().Equal(runnerErrors.NewBadRequestError("credentials %s not defined", s.Fixtures.CreateOrgParams.CredentialsName), err)
+}
+
+func (s *OrgTestSuite) TestCreateOrganizationAlreadyExists() {
+	s.Fixtures.CreateOrgParams.Name = "test-org-1" // this is already created in `SetupTest()`
+
+	_, err := s.Runner.CreateOrganization(s.Fixtures.AdminContext, s.Fixtures.CreateOrgParams)
+
+	s.Require().Equal(runnerErrors.NewConflictError("organization %s already exists", s.Fixtures.CreateOrgParams.Name), err)
+}
+
+func (s *OrgTestSuite) TestCreateOrganizationPoolMgrFailed() {
+	s.Fixtures.PoolMgrCtrlMock.On("CreateOrgPoolManager", s.Fixtures.AdminContext, mock.AnythingOfType("params.Organization"), s.Fixtures.Providers, s.Fixtures.Store).Return(s.Fixtures.PoolMgrMock, s.Fixtures.ErrMock)
+
+	_, err := s.Runner.CreateOrganization(s.Fixtures.AdminContext, s.Fixtures.CreateOrgParams)
+
+	s.Fixtures.PoolMgrMock.AssertExpectations(s.T())
+	s.Fixtures.PoolMgrCtrlMock.AssertExpectations(s.T())
+	s.Require().Equal(fmt.Sprintf("creating org pool manager: %s", s.Fixtures.ErrMock.Error()), err.Error())
+}
+
+func (s *OrgTestSuite) TestCreateOrganizationStartPoolMgrFailed() {
+	s.Fixtures.PoolMgrMock.On("Start").Return(s.Fixtures.ErrMock)
+	s.Fixtures.PoolMgrCtrlMock.On("CreateOrgPoolManager", s.Fixtures.AdminContext, mock.AnythingOfType("params.Organization"), s.Fixtures.Providers, s.Fixtures.Store).Return(s.Fixtures.PoolMgrMock, nil)
+	s.Fixtures.PoolMgrCtrlMock.On("DeleteOrgPoolManager", mock.AnythingOfType("params.Organization")).Return(s.Fixtures.ErrMock)
+
+	_, err := s.Runner.CreateOrganization(s.Fixtures.AdminContext, s.Fixtures.CreateOrgParams)
+
+	s.Fixtures.PoolMgrMock.AssertExpectations(s.T())
+	s.Fixtures.PoolMgrCtrlMock.AssertExpectations(s.T())
+	s.Require().Equal(fmt.Sprintf("starting org pool manager: %s", s.Fixtures.ErrMock.Error()), err.Error())
+}
+
+func (s *OrgTestSuite) TestListOrganizations() {
+	orgs, err := s.Runner.ListOrganizations(s.Fixtures.AdminContext)
+
+	s.Require().Nil(err)
+	s.Require().Equal(len(orgs), len(s.Fixtures.StoreOrgs))
```

Comment:
It may be useful to test that the expected names of the orgs match. Especially if there are only a couple of them. If this check is used in multiple tests, you can write a function to assert they match.

But this is not strictly necessary at this point.